### PR TITLE
GCP infrastructure using terraform

### PR DESCRIPTION
**Pull request by MrManning**
Adds a basic setup for infrastructure

### CODE REVIEW

**Comment by matthew-green-nandos** [comment]
Hey @MrManning, can pull on the upstream master to incorporate PR that have been merged into master.

**Comment by MrManning** [comment]
> Hey @MrManning, can pull on the upstream master to incorporate PR that have been merged into master.

Thanks for letting me know, I have just merged master into my fork

**Comment by MrManning** [comment]
Closed by accident :(

File: infrastructure/gcp/environments/preview/backend.tf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+terraform {
+  backend "gcs" {
+    bucket = "PROJECT_ID-tfstate"
+    prefix = "env/preview"
+  }
+}
+
```
Comment (matthew-green-nandos) [inline infrastructure/gcp/environments/preview/backend.tf]: Hey, probably some whitespace that is causing github to say there is no new line.

File: infrastructure/gcp/environments/preview/main.tf [hunk 2]
```diff
@@ -0,0 +1,17 @@
+provider "google" {
+  project = var.project
+  region  = var.region
+}
+
```
Comment (matthew-green-nandos) [inline infrastructure/gcp/environments/preview/main.tf:4]: I think you might be missing the web infrastructure set up.

File: website/package-lock.json [hunk 3]
```diff
@@ -1,7 +1,13939 @@
 {
```
Comment (matthew-green-nandos) [inline website/package-lock.json]: please exclude this file from this PR as it seems unrelated to terraform.

**Review by matthew-green-nandos — CHANGES_REQUESTED** [review]

**Review by matthew-green-nandos — APPROVED** [review]

**Pull request merged**